osf/models/user.py

```diff
@@ -646,7 +646,7 @@ def osf_groups(self):
 
     def is_institutional_admin(self, institution):
         group_name = institution.format_group('institutional_admins')
-        return self.groups.filter(name=group_name).exists()
+        return self.groups.filter(name=group_name).exists() and institution.can_request_access
```

Comment:
It's difficult question where this AND check should be in the codebase. The `institutional_admins` permission may have other uses in the future that don't involve this feature, even though this is a quick fix and probably works for now you should move this check to the permissions classes that also check for this feature's request_type. `UserMessagePermissions` and `InstitutionalAdminRequestTypePermission` are better places to insert a conditional for this feature.  `is_institutional_admin` can't "lie" about whether the user has that permission just because the feature is on/off.